src/statsmodels/contrasts.jl

```diff
@@ -0,0 +1,233 @@
+## Specify contrasts for coding categorical data in model matrix. Contrasts types
+## are a subtype of AbstractContrasts. ContrastsMatrix types hold a contrast
+## matrix, levels, and term names and provide the interface for creating model
+## matrix columns and coefficient names.
+##
+## Contrasts types themselves can be instantiated to provide containers for
+## contrast settings (currently, just the base level).
+##
+## ModelFrame will hold a Dict{Symbol, ContrastsMatrix} that maps column
+## names to contrasts.
+##
+## ModelMatrix will check this dict when evaluating terms, falling back to a
+## default for any categorical data without a specified contrast.
+##
+## TODO: implement contrast types in Formula/Terms
+##
+## Dave Kleinschmidt 2015-2016
+
+
+"""
+Interface to describe contrast coding schemes for categorical variables.
+
+Concrete subtypes of `AbstractContrasts` describe a particular way of converting a
+categorical column in a `DataFrame` into numeric columns in a `ModelMatrix`. Each
```

Comment:
I would remove `in a DataFrame` here: I think at some point we should move this into StatsBase since it's not specific to DataFrames at all.